### PR TITLE
build: auto download lnd-binary on install

### PR DESCRIPTION
**Pull request by mrfelton**
Use `lnd-binary` to download the correct mac/windows/linux lnd binary into the relevant location in the resources directory as part of the install process.

See https://github.com/mrfelton/lnd-binary

This has a couple of advantages:

1. We make the installtion process easier for users
2. We ensure that users always have the version of lnd that we expect/requite
3. `lnd-binary` takes extra steps to verify the checksum of the downloaded binary to ensure its integrity
4. downloaded binary is cached by npm so future builds are faster

In this version of the PR, I download the binary for all 3 supported operating systems, which is useful for packaging purposes. However, we might want to instead only download the one for the users current platform. Thoughts?

### CODE REVIEW

**Comment by dmp1ce** [comment]
I tried this change today and received the following errors.

```
yarn install v1.7.0
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
info fsevents@1.2.4: The platform "linux" is incompatible with this module.
info "fsevents@1.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning " > eslint-config-airbnb@17.0.0" has incorrect peer dependency "eslint@^4.19.1".
warning "eslint-config-airbnb > eslint-config-airbnb-base@13.0.0" has incorrect peer dependency "eslint@^4.19.1".
warning " > eslint-plugin-compat@2.4.0" has incorrect peer dependency "eslint@^3.0.0 || ^4.0.0".
[5/5] Building fresh packages...
[1/5] ⠐ electron
[-/5] ⠐ waiting...
[3/5] ⠐ lnd-binary
[4/5] ⠐ node-sass
error /home/david/Projects/aur/zap-desktop-standalone-git/src/zap-desktop-standalone-git/node_modules/lnd-binary: Command failed.
Exit code: 1
Command: postinstall-build dist && node ./dist/bin/lnd-install
Arguments:
Directory: /home/david/Projects/aur/zap-desktop-standalone-git/src/zap-desktop-standalone-git/node_modules/lnd-binary
Output:
info lnd-binary Downloading https://github.com/lightningnetwork/lnd/releases/download/v0.4.2-beta/lnd-linux-amd64-v0.4.2-beta.tar.gz
info lnd-binary Verified checksum of downloaded file
info lnd-binary Extracted lnd archive to /tmp/tmp/lnd-downloads-e4e5a368
fs.js:119
    throw err;
    ^

Error: EXDEV: cross-device link not permitted, rename '/tmp/tmp/lnd-downloads-e4e5a368/lnd-linux-amd64-v0.4.2-beta/lnd' -> '/home/david/Projects/aur/zap-desktop-standalone-git/src/zap-desktop-standalone-git/node_modules/lnd-binary/vendor/lnd'
    at Object.renameSync (fs.js:598:3)
    at moveToDest (/home/david/Projects/aur/zap-desktop-standalone-git/src/zap-desktop-standalone-git/node_modules/lnd-binary/dist/lib/extract.js:79:23)
    at Extract.stream.pipe.pipe.on (/home/david/Projects/aur/zap-desktop-standalone-git/src/zap-desktop-standalone-git/node_modules/lnd-binary/dist/lib/extract.js:88:110)
    at Extract.emit (events.js:187:15)
    at finishMaybe (/home/david/Projects/aur/zap-desktop-standalone-git/src/zap-desktop-standalone-git/node_modules/readable-stream/lib/_stream_writable.js:630:14)
    at /home/david/Projects/aur/zap-desktop-standalone-git/src/zap-desktop-standalone-git/node_modules/readable-stream/lib/_stream_writable.js:608:5
```

**Comment by mrfelton** [comment]
@dmp1ce This was an issue in lnd-binary, which I have now resolved and released an updated version of (https://github.com/mrfelton/lnd-binary/releases/tag/v0.3.1). I have updated this PR to use the new lnd-binary release.

**Comment by JimmyMow** [comment]
Tested ACK 689d971

File: app/lnd/config/index.js [hunk 1]
```diff
@@ -51,8 +52,9 @@ switch (plat) {
     break
 }
 
-if (process.env.NODE_ENV === 'development') {
-  lndPath = join(appRootPath, 'resources', 'bin', plat, lndBin)
+if (isDev) {
+  const lndBinaryDir = dirname(require.resolve('lnd-binary/package.json'))
+  lndPath = lndPath = join(lndBinaryDir, 'vendor', lndBin)
```
Comment (Empact) [inline app/lnd/config/index.js]: nit: redundant assignment: `lndPath = lndPath = `

File: app/lnd/config/index.js [hunk 2]
```diff
@@ -51,8 +52,9 @@ switch (plat) {
     break
 }
 
-if (process.env.NODE_ENV === 'development') {
-  lndPath = join(appRootPath, 'resources', 'bin', plat, lndBin)
+if (isDev) {
```
Comment (Empact) [inline app/lnd/config/index.js]: Can you explain why this change?

**Comment by Empact** [comment]
utACK b2e65f2

File: app/lnd/config/index.js (hunk 2 above)
Comment (mrfelton) [reply app/lnd/config/index.js]: When you run `npm start` for example, we run the app with `NODE_ENV=production` so in this case that would trigger the second condition which would result in it trying to load the binary from the resources directory which is not what we want.

`isDev` (https://github.com/sindresorhus/electron-is-dev) is better able to detect wether you are running in development or not. NODE_ENV==='development' !== electron running in development.

**Comment by Empact** [comment]
Sorry fat fingers :P

**Comment by JimmyMow** [comment]
Re-Tested ACK 953d3a4

🎊 🎉
